src/util/rocksdb/mod.rs

```diff
@@ -341,6 +341,33 @@ pub fn compact_range(
     db.compact_range_cf_opt(handle, &compact_opts, start_key, end_key);
 }
 
+pub fn compact_files(db: &DB, cf: &CFHandle, output_level: u32) -> Result<(), String> {
+    let cf_opts = db.get_options_cf(cf);
+    let compressions = cf_opts.get_compression_per_level();
+
+    let mut opts = CompactionOptions::new();
+    if let Some(compression) = compressions.get(output_level as usize) {
+        opts.set_compression(*compression);
+    }
+    opts.set_output_file_size_limit(cf_opts.get_target_file_size_base() as usize);
+
+    let mut input_files = Vec::new();
+    let cf_meta = db.get_column_family_meta_data(cf);
```

Comment:
We should disable auto compaction and confirm there is no compaction is running before get all files.